File: backend/routes/frames.js
import { framesCollection } from "../db.js";
import express from "express";

let userData = {};

export const FramesRoute = express.Router();

FramesRoute.get("/", async (req, res) => {

    try {

        userData.Shape = await framesCollection.distinct("shape");
        userData.Color = await framesCollection.distinct("color");
        userData.Type = await framesCollection.distinct("type");
        userData.Mark = await framesCollection.distinct("mark");

        res.json(userData);

    } catch (error) {

        throw new error;

    }

})

File: backend/routes/frames.py
from fastapi import APIRouter, HTTPException
from db import frames_collection
router = APIRouter()

@router.get("/frames")

async def get_frames():
    try:
        
        framesData = {
            "Shape": await frames_collection.distinct("shape"),
            "Color": await frames_collection.distinct("color"),
            "Type": await frames_collection.distinct("type"),
            "Mark": await frames_collection.distinct("mark")
        }
        
        return framesData
    except Exception as e:
        raise HTTPException(status_code=500, detail=f"Error fetching frames: {str(e)}")
        